launcher/minecraft/ShortcutUtils.cpp

```diff
@@ -0,0 +1,237 @@
+    }
+
+    QIcon iconObj = icon->icon();
+    bool success = iconObj.pixmap(1024, 1024).save(iconPath, "ICNS");
+    iconFile.close();
+
+    if (!success) {
+        iconFile.remove();
+        QMessageBox::critical(shortcut.parent, QObject::tr("Create Shortcut"), QObject::tr("Failed to create icon for application."));
+        return;
+    }
+#elif defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
+    if (appPath.startsWith("/tmp/.mount_")) {
+        // AppImage!
+        appPath = QProcessEnvironment::systemEnvironment().value(QStringLiteral("APPIMAGE"));
+        if (appPath.isEmpty()) {
+            QMessageBox::critical(
+                shortcut.parent, QObject::tr("Create Shortcut"),
+                QObject::tr("Launcher is running as misconfigured AppImage? ($APPIMAGE environment variable is missing)"));
+        } else if (appPath.endsWith("/")) {
+            appPath.chop(1);
+        }
+    }
+
+    iconPath = FS::PathCombine(shortcut.instance->instanceRoot(), "icon.png");
+
+    QFile iconFile(iconPath);
+    if (!iconFile.open(QFile::WriteOnly)) {
+        QMessageBox::critical(shortcut.parent, QObject::tr("Create Shortcut"), QObject::tr("Failed to create icon for shortcut."));
+        return;
+    }
+    bool success = icon->icon().pixmap(64, 64).save(&iconFile, "PNG");
+    iconFile.close();
+
+    if (!success) {
+        iconFile.remove();
+        QMessageBox::critical(shortcut.parent, QObject::tr("Create Shortcut"), QObject::tr("Failed to create icon for shortcut."));
+        return;
+    }
+
+    if (DesktopServices::isFlatpak()) {
+        appPath = "flatpak";
+        args.append({ "run", BuildConfig.LAUNCHER_APPID });
+    }
+
+#elif defined(Q_OS_WIN)
+    iconPath = FS::PathCombine(shortcut.instance->instanceRoot(), "icon.ico");
+
+    // part of fix for weird bug involving the window icon being replaced
+    // dunno why it happens, but parent 2-line fix seems to be enough, so w/e
+    auto appIcon = APPLICATION->getThemedIcon("logo");
+
+    QFile iconFile(iconPath);
+    if (!iconFile.open(QFile::WriteOnly)) {
+        QMessageBox::critical(shortcut.parent, QObject::tr("Create Shortcut"), QObject::tr("Failed to create icon for shortcut."));
+        return;
+    }
+    bool success = icon->icon().pixmap(64, 64).save(&iconFile, "ICO");
+    iconFile.close();
+
+    // restore original window icon
+    QGuiApplication::setWindowIcon(appIcon);
+
+    if (!success) {
+        iconFile.remove();
+        QMessageBox::critical(shortcut.parent, QObject::tr("Create Shortcut"), QObject::tr("Failed to create icon for shortcut."));
+        return;
+    }
+
+#else
+    QMessageBox::critical(shortcut.parent, QObject::tr("Create Shortcut"), QObject::tr("Not supported on your platform!"));
+    return;
+#endif
+    args.append({ "--launch", shortcut.instance->id() });
+    args.append(shortcut.extraArgs);
+
+    if (!FS::createShortcut(std::move(filePath), appPath, args, shortcut.name, iconPath)) {
```

Comment:
Nit: it's not possible to move a const reference.

```suggestion
    if (!FS::createShortcut(filePath, appPath, args, shortcut.name, iconPath)) {
```

Reply:
Ah, somehow CLion forgot to warn about that...
Fixed now